Resubmit 6U control plane + system mesh changes

aliuTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/19037

Problem description
We are saying goodbye to relying on ethernet coordinates because this won't be supported for 6U galaxy and blackhole. Instead, we source queries on "physical coordinates" through our new shiny ControlPlane.

What's changed
Deprecate old static configuration files used to provide a logical->physical mapping of coordinates in favor of queries to ControlPlane.
Simplify SystemMesh in how it now maintains this mapping.

Resubmission of https://github.com/tenstorrent/tt-metal/pull/19096, with one n150 fix

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

aliuTT committed:
Revert "Revert 6U Mesh changes because it broke a bunch of N150 tests (#19339)"

This reverts commit 01fb2366ca8078a581a0ad258dc6e33b45b76050.

omilyutin-tt reviewed (commented):
Thanks!!

There are changes to umd / tt_llk - intentional?

aliuTT committed:
Patch missing handling of n150

aliuTT committed:
Fix TG unit tests

aliuTT committed:
Update system mesh logic to rotate physical mesh if needed

# Requesting a (8x4) mesh on a system like this
# 11 04 05 18 19 24 25 30
# 10 07 06 17 20 23 26 29
# 09 08 15 16 21 22 27 28
# 12 13 14 35 34 33 32 31
# Mesh was incorrectly returning
# Row 0 -> 11 04 05 18
# Row 1 -> 19 24 25 30
# Row 2 -> 10 07 06 17
# Row 3 -> 20 23 26 29
# etc.

bbradelTT reviewed (approved):
Please look into the GitHub Actions errors. It would be good if they didn't show up.

cfjchu commented on tt_metal/distributed/system_mesh.cpp:
```diff
@@ -169,31 +143,35 @@ std::vector<chip_id_t> SystemMesh::Impl::get_mapped_physical_device_ids(const Me
+        }
+    } else {
+        for (const auto& system_coord : system_range) {
+            auto physical_device_id = get_physical_device_id(MeshCoordinate(system_coord[1], system_coord[0]));
```
@aliuTT we should just retain the existing code here for the non-rotated case:

`auto physical_device_id = get_physical_device_id(system_coord);`

aliuTT replied:
copy paste error probably

cfjchu commented on tt_metal/distributed/system_mesh.cpp:
```diff
@@ -169,31 +143,35 @@ std::vector<chip_id_t> SystemMesh::Impl::get_mapped_physical_device_ids(const Me
+                auto system_coord = MeshCoordinate(j, i);
+                auto physical_device_id = get_physical_device_id(system_coord);
+                physical_device_ids.push_back(physical_device_id);
+                log_info(LogMetal, "Logical coordinate: {}, Physical device ID: {}", system_coord, physical_device_id);
```
flip this to `log_debug`

aliuTT committed:
Try fixing test

cfjchu committed:
Fix mesh shape transpose & failing cases
